airavata-api/airavata-api-server/src/main/java/org/apache/airavata/api/server/handler/AiravataServerHandler.java

```diff
@@ -692,6 +695,80 @@ public String generateAndRegisterSSHKeys(AuthzToken authzToken, String descripti
         }
     }
 
+    @Override
+    public void validateStorageLimit(AuthzToken authzToken, ExperimentModel experiment, String storageResourceId)
```

Comment:
This method is a very risky one. You can see exceptions at most of the lines. You can use fine grained exception handling by wrapping each sub section with try catches to provide more debug information